dev/io.openliberty.threading.virtual.internal/src/io/openliberty/threading/virtual/internal/VirtualThreadOperations.java

```diff
@@ -63,6 +82,9 @@ public boolean isVirtual(Thread thread) {
 
     @Override
     public boolean isSupported() {
-        return true;
+        if (overrideService != null)
+            return overrideService.allowVirtualThreadCreation();
+        else
+            return true;
```

Comment:
VirtualThreadOps provides more than just virtual thread creation. It is a general abstraction for all Java SE methods that were introduced for virtual threads.  Liberty components will use it to ask whether or not a Thread is virtual.  This is needed even if Liberty itself does not ever create virtual threads because the application can create virtual threads regardless.  The `isSupported` method is supposed to be returning whether the Java level used supports virtual threads.  You need a different method if you want to expose whether or not Liberty is allowed to create virtual threads.

```suggestion
        return true;
    }

    @Override
    public boolean isVirtualThreadCreationEnabled() {
        ThreadTypeOverride override = overrideService;
        return override == null ? true : override.allowVirtualThreadCreation();
```

Reply:
latest commit resolves this